### PR TITLE
PP-5388: salient event external state map

### PR DESCRIPTION
**Pull request by sfount**
* add missing external states
* add latest version of events and their map to states

Depends on https://github.com/alphagov/pay-connector/pull/1503

### CODE REVIEW

File: src/main/java/uk/gov/pay/ledger/event/model/SalientEventType.java [hunk 1]
```diff
@@ -5,7 +5,32 @@
 
 public enum SalientEventType {
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/ledger/event/model/SalientEventType.java]: I wonder if we should rename this to `EventType` - feel like the salient concept has died

File: src/main/java/uk/gov/pay/ledger/transaction/state/TransactionState.java [hunk 2]
```diff
@@ -46,10 +54,34 @@ public String getMessage() {
         return message;
     }
 
-    private static final Map<SalientEventType, TransactionState> EVENT_TYPE_TRANSACTION_STATE_MAP = Map.of(
-            SalientEventType.PAYMENT_CREATED, CREATED,
-            SalientEventType.AUTHORISATION_SUCCESSFUL, SUBMITTED
-    );
+    private static final Map<SalientEventType, TransactionState> EVENT_TYPE_TRANSACTION_STATE_MAP =
+            ImmutableMap.<SalientEventType, TransactionState>builder()
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/ledger/transaction/state/TransactionState.java]: `Map.of` returns an immutable map (`ImmutableMap` was a google util for pre Java 11 times)

File: src/main/java/uk/gov/pay/ledger/event/model/SalientEventType.java (hunk 1 above)
Comment (sfount) [reply src/main/java/uk/gov/pay/ledger/event/model/SalientEventType.java]: @heathd suggested just using Strings in this map as the contract with Connector events is based on their name and the object isn't used anywhere else, what do you think? 

Happy to rename to `EventType` in the interim

File: src/main/java/uk/gov/pay/ledger/transaction/state/TransactionState.java (hunk 2 above)
Comment (sfount) [reply src/main/java/uk/gov/pay/ledger/transaction/state/TransactionState.java]: `Map.of` has a size limit for the number of pairs. As you've suggested I'll use `Map.ofEntries` in favour of an external library

**Comment by sfount** [comment]
@mrlumbu 
https://github.com/alphagov/pay-ledger/pull/114/commits/df5b434eebfb3e13f25984788dd56d4bb589d82b
* `SalientEventType` -> `EventType` 
* `ImmutableMap` -> `Map.ofEntries`

**Pull request merged**